crates/runtime/src/dataconnector.rs

```diff
@@ -138,10 +142,32 @@ pub async fn create_new_connector(
 
     let connector_factory = guard.get(name);
 
-    match connector_factory {
-        Some(factory) => Some(factory(secret, params).await),
-        None => None,
+    if let Some(create_future) = connector_factory {
+        match create_future(secret, params).await {
+            Ok(connector) => return Some(Ok(connector)),
+            Err(err) => {
+                if let Some(err_source) = err.source() {
+                    #[cfg(feature = "postgres")]
+                    if let Some(postgrespool::Error::InvalidHostOrPortError {
```

Comment:
Just wonder if this level of details can be pushed down into the postgres.rs.

Reply:
Ah yeah, this is a bad pattern to set imo. We should have the dataconnector abstraction be unaware of the specific implementations.

Reply:
@phillipleblanc I don't think we should to put spicepod specific errors into pool implementation

Reply:
updated error messages to be out of spicepod scope, and moved outside of dataconnector